compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt

```diff
@@ -445,15 +448,22 @@ class ComposeScene internal constructor(
                 focusedOwner?.onDismissRequest?.invoke()
             } else {
                 currentOwner.processPointerInput(event)
+                mousePressOwner = currentOwner
             }
         } else {
             focusedOwner?.processPointerInput(event)
+            mousePressOwner = focusedOwner
         }
     }
 
     private fun onMouseReleased(event: PointerInputEvent) {
-        val owner = hoveredOwner ?: focusedOwner
-        owner?.processPointerInput(event)
+        if (wasMouseDragEvent) {
```

Comment:
Looks somewhat fragile, as now it logic of who and when get an event become really unclear. Could you describe intended behavior in text?

Reply:
Probably we need to implement simple logic as with two JFrame. In that case, we need not only send Release event into `mousePressOwner`, but any event.

Check this:
```
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch
import kotlinx.coroutines.runBlocking
import kotlinx.coroutines.swing.Swing
import java.awt.Dimension
import java.awt.event.MouseAdapter
import java.awt.event.MouseEvent
import java.awt.event.MouseMotionListener
import javax.swing.JFrame

fun main() {
    runBlocking(Dispatchers.Swing) {
        JFrame().apply {
            addMouseMotionListener(object : MouseMotionListener {
                override fun mouseDragged(e: MouseEvent?) = Unit
                override fun mouseMoved(e: MouseEvent?) = println("mouseMoved")
            })
            addMouseListener(object: MouseAdapter() {
                override fun mouseReleased(e: MouseEvent?) = println("mouseReleased")
            })
            addMouseWheelListener {
                println("mouseWheelMoved")
            }
            size = Dimension(500, 500)
            isVisible = true
        }
        JFrame().apply {
            addMouseMotionListener(object : MouseMotionListener {
                override fun mouseDragged(e: MouseEvent?) = Unit
                override fun mouseMoved(e: MouseEvent?) = println("mouseMoved2")
            })
            addMouseListener(object: MouseAdapter() {
                override fun mouseReleased(e: MouseEvent?) = println("mouseReleased2")
            })
            addMouseWheelListener {
                println("mouseWheelMoved2")
            }
            size = Dimension(500, 500)
            isVisible = true
        }
    }
}
```

When we press in the first jframe, it captures mouse focus, and all mouse events  will be send only to it, not to the second jframe.

`wasMouseDragEvent` here can stuck, if we miss the release event (for example press mouse, and minimize the window)

Maybe we just replace `hoveredOwner` by mouseOwner? And add some logic to change it.

Reply:
Done.